docs/react/guides/advanced-ssr.md

```diff
@@ -0,0 +1,390 @@
+---
+id: advanced-ssr
+title: Advanced Server Rendering
+---
+
+Welcome to the Advanced Server Rendering guide, where you will learn all about using React Query with streaming, Server Components and the Next.js app router.
+
+You might want to read the [Server Rendering & Hydration guide](../guides/ssr) before this one as it teaches the basics for using React Query with SSR, and [Performance & Request Waterfalls](../guides/request-waterfalls) as well as [Prefetching & Router Integration](../guides/prefetching) also contains valuable background.
+
+We'll be using `useSuspenseQuery` here, so also check out the [Suspense guide](../guides/suspense).
+
+Before we start, let's note that while the `initialData` approach outlined in the SSR guide also works with Server Components, we'll focus this guide on the hydration APIs.
+
+## Server Components & Next.js app router
+
+We won't cover Server Components in depth here, but the short version is that they are components that are guaranteed to always run on the server, not only on the initial page view but **also on page transitions**. This is similar to how Next.js `getServerSideProps`/`getStaticProps` and Remix `loader` works, as these also always run on the server but while those can only return data, Server Components can do a lot more. The data part is central to React Query however, so let's focus on that.
```

Comment:
Not sure how to word this but I think we should mention that server components **only** run on the server (and never on the client). It reads a bit weird if we do:

> that are guaranteed to always **only** run on the server, not only ...

Reply:
Very true, current wording is confusing. I'll reword it!